### PR TITLE
Mixerless metric label parity

### PR DESCRIPTION
**Pull request by bianpengyuan**
Proxy part of https://github.com/istio/istio/issues/17231

After this PR, metric become:
```
envoy_requests_total{ 
   response_code="200",
   reporter="source",
   source_workload="productpage-v1",
   source_workload_namespace="default",
   source_principal="unknown",
   source_app="productpage",
   source_version="v1",
   destination_workload="reviews-v1",
   destination_workload_namespace="default",
   destination_principal="unknown",
   destination_app="reviews",
   destination_version="v1",
   destination_service="reviews.default.svc.cluster.local",
   destination_service_name="reviews",
   destination_service_namespace="default",
   request_protocol="http",
   response_flags="-",
   connection_security_policy="unknown",
   permissive_response_code="none",
   permissive_response_policyid="none"
}
```

### CODE REVIEW

File: extensions/common/context.cc [hunk 1]
```diff
@@ -42,12 +43,37 @@ using Envoy::Extensions::Common::Wasm::Null::Plugin::getValue;
...
 namespace Common {
 
+const char kRbacFilterName[] = "envoy.filters.http.rbac";
+const char kRbacPermissivePolicyIDField[] = "shadow_effective_policy_id";
+const char kRbacPermissiveEngineResultField[] = "shadow_engine_result";
+
+namespace {
+
+// Extract fqdn from Istio cluster name. If cluster name does not follow Istio
+// convention, fqdn will be left as empty string.
```
Comment (mandarjog) [inline extensions/common/context.cc]: 1. Please add documentation and example of pilot generated cluster names.
2. Also add tests for the same.

Comment (mandarjog) [reply extensions/common/context.cc]: In addition to cluster name, this is the cluster metadata for TD.
Use it as a 2nd level fall back.

``` c++
getStringValue({"cluster_metadata", "filter_metadata", "com.google.trafficdirector", "backend_service_name"}, &service_name);
```
```yaml
    "filter_metadata":
        "com.google.trafficdirector":
              "backend_service_name": "csm-serve404"
```

Comment (mandarjog) [reply extensions/common/context.cc]: scrap that. Do not read from cluster metadata yet.

Comment (bianpengyuan) [reply extensions/common/context.cc]: Done. Update SD e2e test a bit to reflect this change. It is a bit of hard to add unit test due to usage of wasm API.

**Pull request merged**